var/spack/repos/builtin/packages/navi/package.py

```diff
@@ -0,0 +1,21 @@
+# Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import *
+
+
+class Navi(MakefilePackage):
+    """An interactive cheatsheet tool for the command-line"""
+
+    homepage = "https://github.com/denisidoro/navi"
+    url = "https://github.com/denisidoro/navi/archive/refs/tags/v2.20.1.tar.gz"
+
+    maintainers = ["denisidoro"]
+
+    version("2.20.1", sha256="92644677dc46e13aa71b049c5946dede06a22064b3b1834f52944d50e3fdb950")
+
+    def install(self, spec, prefix):
+        cargo = which("cargo")
```

Comment:
Looks like you need a dependency on `cargo` added above (below the `version` directive).

Reply:
But that also means you need to add a package for installing it as well.

Reply:
thanks, I've added the dependency to the file

Reply:
Not everyone who may want to build this package will have `cargo` installed and no-one has added the package for it.

Adding the dependency is the first step but there still needs to be a package added for `cargo`.

Reply:
But I've added the dependency on https://github.com/spack/spack/pull/32895/commits/299e3954bd8fcefdb13944881b81c2005c711b7e 🤔

Reply:
sorry, I forgot that `cargo` is installed with `rust`.

Just changed it